packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,124 @@
+import { attr, FASTElement } from '@microsoft/fast-element';
+
+/**
+ * A Tooltip Custom HTML Element.
+ * Based on ARIA APG Tooltip Pattern {@link https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/ }
+ * @public
+ */
+export class Tooltip extends FASTElement {
+  /**
+   * The attached element internals
+   */
+  public elementInternals = this.attachInternals();
+
+  /**
+   * Set the delay for the tooltip
+   */
+  @attr
+  public delay?: number;
+
+  /**
+   * The default delay for the tooltip
+   * @internal
+   */
+  private defaultDelay: number = 250;
+
+  /**
+   * Set the positioning of the tooltip
+   */
+  @attr
+  public positioning?: string;
+
+  /**
+   * The id of the anchor element for the tooltip
+   */
+  @attr
+  public anchor: string = '';
+
+  /**
+   * Reference to the anchor element
+   * @internal
+   */
+  private get anchorElement(): HTMLElement | null {
+    return document.getElementById(this.anchor ?? '');
+  }
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.anchorElement) {
+      this.anchorElement.setAttribute('aria-describedby', `tooltip-${this.anchor}`);
```

Comment:
Same question as with the template: what happens if the `id` of the tooltip is overridden?

Reply:
Now uses a unique ID if none provided. Follows the pattern used in Accordion.

```
  @attr
  public id: string = uniqueId('tooltip-');
  ```

Then I made the anchor element ID more dynamic as well. 

```
const anchorName = this.anchorElement.style.anchorName || `--${this.anchor}`;
```